src/org/scada_lts/web/mvc/api/components/is_alive/TimeFromServer.java

```diff
@@ -42,4 +47,89 @@
             return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
         }
     }
+
+    /**
+     * @author radek2s <rjajko@softq.pl>
+     * @return ResponseEntity with status message
+     */
+    @GetMapping(value = "/time2")
+    public ResponseEntity<Long> getServerTime(HttpServletRequest request) {
+        LOG.info("/api/is_alive/time2");
+        try {
+            User user = Common.getUser(request);
+            if (user != null) {
+                long unixTimestamp = System.currentTimeMillis();
+                return new ResponseEntity<>(unixTimestamp, HttpStatus.OK);
+            }
+            return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
+        } catch (Exception e) {
+            return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    /**
+     * @author radek2s <rjajko@softq.pl>
+     * @return ResponseEntity with status message
+     */
+    @PostMapping(value = "/watchdog")
+    public ResponseEntity<String> notifyWatchdog(@RequestBody WatchDogConfig config, HttpServletRequest request) {
```

Comment:
`@RequestBody(required = true)`

Then if the object is null it will automatically return 400 code, you do not need to add if in the handling method of this endpoint.